src/main/java/seedu/internbuddy/logic/parser/WithdrawCommandParser.java

```diff
@@ -0,0 +1,31 @@
+package seedu.internbuddy.logic.parser;
+
+import static seedu.internbuddy.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.internbuddy.commons.core.index.Index;
+import seedu.internbuddy.logic.commands.WithdrawCommand;
+import seedu.internbuddy.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new WithdrawCommand object
+ */
+public class WithdrawCommandParser implements Parser<WithdrawCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the WithdrawCommand
+     * and returns a WithdrawCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public WithdrawCommand parse(String args) throws ParseException {
+        try {
+            Index[] indexes = ParserUtil.parseIndexes(args);
+            Index companyIndex = indexes[0];
+            Index applicationIndex = indexes[1];
```

Comment:
After discussion, the current behaviour is fine. But I still suggest that `ParserUtil#parseIndexes` be renamed to `ParserUtil#parseWithdrawIndex` to avoid confusion.